### DISCUSSION TITLE
How do I backup all my sessions?

### DISCUSSION THREAD

**Discussion by smashah** [open-wa/wa-automate-nodejs#3096] [General]
It's essential to know how to backup and restore your sessions. This ensures the safety of your data and allows you to resume your work without any hassle. In this post, we'll guide you through the process of backing up your Open-WA session folders without including unnecessary files, making it a lightweight and efficient solution. We will cover the steps for Ubuntu, macOS, and Windows.

Note: This tutorial assumes that you have already set up Open-WA and have a working session folder(s).

## Step 1: Install Required Tools

First, you need to make sure that the required tools, tar and zstd, are installed on your system. These tools will help you compress and archive your session folders.

For Ubuntu, run the following command:

```bash
> apt update && apt install zstd tar -y
```

For macOS, you can install these tools using Homebrew:

```bash
> brew install zstd tar
```
For Windows, you'll need to use a POSIX-compatible environment like [WSL](https://docs.microsoft.com/en-us/windows/wsl/) or [Git Bash](https://gitforwindows.org/), and install the required tools using the package manager for the environment.

## Step 2: Backup Session Folders

Once you have the required tools installed, you can proceed to backup your session folders. In this example, we'll assume that all your session folders are located within a root folder, e.g., /sessions.

For Ubuntu, macOS, and Windows (with WSL or Git Bash), run the following command:

```bash
time find /sessions -type f -name "Local Storage" -o -name 'IndexedDB' | tar --zstd -cvf /sessions/ALL_SESSIONS_BACKUP.tar.zstd --exclude="*.log" --exclude="LOG*" --exclude=".DS_Store" -T -
```

This command will create a compressed archive, ALL_SESSIONS_BACKUP.tar.zstd, containing all session folders within the /sessions directory, while excluding unnecessary files such as logs and .DS_Store files.

## Step 3: Save and Restore the Backup

After creating the backup, you should save the ALL_SESSIONS_BACKUP.tar.zstd file to a secure location, such as an external storage device or a cloud storage service.

When you need to restore your session folders, simply copy the ALL_SESSIONS_BACKUP.tar.zstd file back to the desired location and run the following command:

For Ubuntu, macOS, and Windows (with WSL or Git Bash):

```bash
> tar --use-compress-program=unzstd -xvf ALL_SESSIONS_BACKUP.tar.zstd
```

This will decompress and restore your session folders to their original state.

## Conclusion:

Backing up your Open-WA session folders is a crucial step to ensure the safety and continuity of your work. With the steps outlined in this blog post, you can efficiently backup and restore your session folders without including unnecessary files. Remember to always store your backups in a secure location to prevent data loss.

**Comment by ahmed-safari** [comment]
On macOS, I got the following error
Warning: No available formula with the name "tar". Did you mean tart, star, tag, tal, txr, tor, par, bar or dar?

Fixed by using gnu-tar instead.
